core/course_controller/CourseController.py

```diff
@@ -21,6 +41,13 @@ def save_course(course_data: CourseFormData, course_id: int | None = None) -> st
         if not course_data.course_code:
             raise ValueError("Course code cannot be empty.")
 
+        #Get semester instance using semester name
+        try:
+            semester = Semester.objects.get(semester_name = semester_name)
```

Comment:
same as below. i think this would be handled when getting the course using semester__semester_name